lib/adhearsion/call.rb

```diff
@@ -209,6 +210,12 @@ def register_initial_handlers
         merge_headers event.headers
       end
 
+      register_event_handler Punchblock::Event::Complete do |event|
```

Comment:
You'll want to unit test setting this state based on events by writing tests similar to [these](https://github.com/adhearsion/adhearsion/blob/develop/spec/adhearsion/call_spec.rb#L553-L580).